src/cartesian/getTicks.ts

```diff
@@ -1,81 +1,34 @@
-import _ from 'lodash';
 import { CartesianTickItem, Size } from '../util/types';
 import { mathSign, isNumber } from '../util/DataUtils';
 import { getStringSize } from '../util/DOMUtils';
 import { Props as CartesianAxisProps } from './CartesianAxis';
 import { Global } from '../util/Global';
-import { getEveryNthWithCondition } from '../util/getEveryNthWithCondition';
-import { getAngledRectangleWidth } from '../util/CartesianUtils';
-
-/**
- * Given an array of ticks, find N, the lowest possible number for which every
- * nTH tick in the ticks array isShow == true and return the array of every nTh tick.
- * @param {CartesianTickItem[]} ticks An array of CartesianTickItem with the
- * information whether they can be shown without overlapping with their neighbour isShow.
- * @returns {CartesianTickItem[]} Every nTh tick in an array.
- */
-export function getEveryNThTick(ticks: CartesianTickItem[]) {
-  let N = 1;
-  let previous = getEveryNthWithCondition(ticks, N, tickItem => tickItem.isShow);
-  while (N <= ticks.length) {
-    if (previous !== undefined) {
-      return previous;
-    }
-    N++;
-    previous = getEveryNthWithCondition(ticks, N, tickItem => tickItem.isShow);
-  }
-
-  return ticks.slice(0, 1);
-}
-
-export function getNumberIntervalTicks(ticks: CartesianTickItem[], interval: number) {
-  return getEveryNthWithCondition(ticks, interval + 1);
-}
-
-function getAngledTickWidth(contentSize: Size, unitSize: Size, angle: number) {
-  const size = { width: contentSize.width + unitSize.width, height: contentSize.height + unitSize.height };
-
-  return getAngledRectangleWidth(size, angle);
-}
-
-function getTicksEnd({
-  angle,
-  ticks,
-  tickFormatter,
-  viewBox,
-  orientation,
-  minTickGap,
-  unit,
-  fontSize,
-  letterSpacing,
-}: Omit<CartesianAxisProps, 'tickMargin'>): CartesianTickItem[] {
-  const { x, y, width, height } = viewBox;
-  const sizeKey = orientation === 'top' || orientation === 'bottom' ? 'width' : 'height';
-  // we need add the width of 'unit' only when sizeKey === 'width'
-  const unitSize: Size =
-    unit && sizeKey === 'width' ? getStringSize(unit, { fontSize, letterSpacing }) : { width: 0, height: 0 };
+import {
+  doesTickFitInBetweenStartAndEnd,
+  getEveryNThTick,
+  getInitialStartAndEnd,
+  getNumberIntervalTicks,
+  getSizeOfTick,
+} from '../util/TickUtils';
+
+type Sign = 0 | 1 | -1;
```

Comment:
nit: what is this? from just reading the code I can't really tell

Reply:
Also now that I looked at `mathSign`, it seems like a really silly function? https://github.com/recharts/recharts/blob/master/src/util/DataUtils.ts#L3C14-L12

Reply:
It is a really silly function 🤦 But I need to time box the refactoring and not explode scope. This totally deserves multiple iterations. 

The sign is an implementation detail for whether we position ticks from left to right or from right to left.

Reply:
Makes sense